SU2_CFD/src/output/COutput.cpp

```diff
@@ -0,0 +1,1345 @@
+    }
+  }
+  
+  if (HistoryOutput_Map[Conv_Field].FieldType == TYPE_COEFFICIENT){
+    AddHistoryOutput("CAUCHY", "C["  + HistoryOutput_Map[Conv_Field].FieldName + "]", FORMAT_SCIENTIFIC, "RESIDUAL");
+  }
+  
+   map<string, bool>::iterator it = Average.begin();
+   for (it = Average.begin(); it != Average.end(); it++){
+     AddHistoryOutput("AVG_" + it->first, "avg[" + AverageGroupName[it->first] + "]", FORMAT_FIXED, "AVG_RES");
+   }
+  
+}
+
+bool COutput::WriteScreen_Header(CConfig *config) {  
+  bool write_header = false;
+  if (config->GetUnsteady_Simulation() == STEADY) {
+    write_header = ((curr_InnerIter % (config->GetWrt_Con_Freq()*40)) == 0) || (config->GetMultizone_Problem() && curr_InnerIter == 0);
+  } else if  (config->GetUnsteady_Simulation() == TIME_STEPPING) {
+    if (!config->GetRestart())
+      write_header = ((curr_TimeIter % (config->GetWrt_Con_Freq()*40)) == 0) || (config->GetMultizone_Problem() && curr_InnerIter == 0);    
+    else {
+      write_header = (((curr_TimeIter - config->GetRestart_Iter()) % (config->GetWrt_Con_Freq()*40)) == 0) || (config->GetMultizone_Problem() && curr_InnerIter == 0);    
+    }
+  } else {
+    write_header = (config->GetUnsteady_Simulation() == DT_STEPPING_1ST || config->GetUnsteady_Simulation() == DT_STEPPING_2ND) && config->GetInnerIter() == 0;
+  }
+
+  /*--- For multizone problems, print the header only if requested explicitly (default of GetWrt_ZoneConv is false) ---*/
+  if(config->GetMultizone_Problem()) write_header = (write_header && config->GetWrt_ZoneConv());
+
+  return write_header;
+}
+
+bool COutput::WriteScreen_Output(CConfig *config) {
+  bool write_output = false;
+  
+  write_output = config->GetnInner_Iter() - 1 == curr_InnerIter;
+  
+  if (((config->GetUnsteady_Simulation() == DT_STEPPING_1ST) || (config->GetUnsteady_Simulation() == DT_STEPPING_2ND) )){
+    write_output = write_output || PrintOutput(config->GetInnerIter(), config->GetWrt_Con_Freq_DualTime());
+  }
+  else if (((config->GetUnsteady_Simulation() == STEADY) || (config->GetUnsteady_Simulation() == TIME_STEPPING) )){
+    write_output = write_output ||  PrintOutput(config->GetInnerIter(), config->GetWrt_Con_Freq()) ;    
+  } 
+
+  /*--- For multizone problems, print the body only if requested explicitly (default of GetWrt_ZoneConv is false) ---*/
+  if(config->GetMultizone_Problem()) write_output = (write_output && config->GetWrt_ZoneConv());
+
+  return write_output;
+}
+
+bool COutput::WriteHistoryFile_Output(CConfig *config) { 
+// if (!write_dualtime){
+//   return true;
+// }
+// else {
+//   return false;
+// }
+  return true;
+}
+
+void COutput::SetCommonHistoryFields(CConfig *config){
+  
+  /// BEGIN_GROUP: ITERATION, DESCRIPTION: Iteration identifier.
+  /// DESCRIPTION: The time iteration index.
+  AddHistoryOutput("TIME_ITER",     "Time_Iter",  FORMAT_INTEGER, "ITER"); 
```

Comment:
I had a suggestion to improve the config option tracking/documentation (and keep it automatically up to date), and I think we could try it here too for the new output options that are registered. 

Is it easy for you to add the string description as an input directly into the prototype for AddHistoryOutput() as well as add a print_description() method that would write it to the console (or write all registered options in one loop)? That way, if a user wanted to get the list of all available outputs for a particular solver, we could have a "help" option in the config file / at the command line that could first print all of the group names with their descriptions, followed by all available fields within a chosen group with their string descriptions. Should be easier to maintain and expose, since the descriptions will be required directly in the C++ whenever adding the option. 

We could consider adding a similar self-describing approach for all config options if this works.